src/lightning/fabric/wrappers.py

```diff
@@ -185,6 +189,28 @@ def __getattr__(self, item: Any) -> Any:
             self._validate_method_access(item, attr)
             return attr
 
+    def __setattr__(self, name: str, value: Any) -> None:
+        if not self._fabric_module_initialized:
+            return super().__setattr__(name, value)
+
+        # Get the _original_module attribute
+        original_module = self._original_module
+        original_has_attr = hasattr(original_module, name)
+        # Can't use super().__getattrr__ because nn.Module only checks _parameters, _buffers, and _modules
+        fabric_has_attr = name in self.__dict__
+
+        if not (original_has_attr or fabric_has_attr):
+            return setattr(original_module, name, value)
```

Comment:
I might have missed previous discussions. Why doesn't this set it for both?

If they both define it, we set it for both
If neither of them define it, only in the original.

Why is that?

Reply:
Because if the user does this

```
my_model.new_attribute = 1
```
Their intention is to set the attribute on the original module. It would be pointless to set a new attribute on our fabric wrapper and also on the original, because there is no mechanism to keep it in sync, and our getattr function would potentially return the wrong value.